vitessce/wrappers.py

```diff
@@ -461,14 +466,362 @@ def __init__(self, loom, **kwargs):
     def get_cells(self, port, dataset_uid, obj_i):
         obj_routes = []
         obj_file_defs = []
-        """
-        try:
-            import loompy
-            if type(self.loom) == loompy.LoomConnection:
-                pass
-                # TODO: append routes
-                # TODO: add file definitions
-        except ImportError:
-            pass
-        """
+
+        # TODO: append routes
+        # TODO: add file definitions
+ 
+        return obj_file_defs, obj_routes
+
+class SnapWrapper(AbstractWrapper):
+
+    # The Snap file is difficult to work with.
+    # For now we can use the processed cell-by-bin MTX file
+    # However, the HuBMAP pipeline currently computes this with resolution 5000
+    # TODO: Make a PR to sc-atac-seq-pipeline to output this at a higher resolution (e.g. 200)
+    # https://github.com/hubmapconsortium/sc-atac-seq-pipeline/blob/develop/bin/snapAnalysis.R#L93
+
+    def __init__(self, in_mtx, in_barcodes_df, in_bins_df, in_clusters_df, starting_resolution=5000):
+        self.in_mtx = in_mtx # scipy.sparse.coo.coo_matrix (filtered_cell_by_bin.mtx)
+        self.in_barcodes_df = in_barcodes_df # pandas dataframe (barcodes.txt)
+        self.in_bins_df = in_bins_df # pandas dataframe (bins.txt)
+        self.in_clusters_df = in_clusters_df # pandas dataframe (umap_coords_clusters.csv)
+
+        self.tempdir = tempfile.mkdtemp()
+
+        self.starting_resolution = starting_resolution
+
+        # Convert to dense matrix if sparse.
+        if type(in_mtx) == coo_matrix:
+            self.in_mtx = in_mtx.toarray()
+
+
+    def create_genomic_multivec_zarr(self, zarr_filepath):
```

Comment:
Can this be refactored to use a `ChromZarr` entity?  If there is no use case outside of SnapTools for this sort of thing, I can understand but if other ATAC-seq data sets or other assays use this sort of thing, it might be worth pulling this out in bulk.

Reply:
I am not really sure whether other assays would use similar code. I would hope that other formats would be much easier to work with than the Snap files. Standard files like BigWig should not require this much conversion code.

Reply:
But we would still want a standardized way of writing out data, like an api `chrom_zarr = ChromZarr()` and `chrom_zarr.add_data()`?